### ISSUE TITLE
Windows compile

### ISSUE THREAD

**Issue by jstaursky** [howl-editor/howl#309]
Im trying to build this on windows using MSYS2 and getting  close but it stops when make tries to build lpeg.o, various sources seem to say that lpeg can only be built as a dll on windows (maybe also as a .lib but I have not seen this). Otherwise I have all the build requirements setup (MSYS2 provides libluajit-5.1.dll.a as libluajit.a). What is the remedy or plan for this? If there is none, when is the last commit to the windows branch that worked? I'll just clone that and be happy :)

**Comment by kirbyfan64** [comment]
Wait, so you were using the `windows` branch? Did you follow *exactly* [these](https://github.com/howl-editor/howl/issues/201#issuecomment-223068288) instructions? What were the exact error messages?

**Comment by jstaursky** [comment]
Thanks for the tip! While instructions did generate an executable no matter what I do it fails to run. The closest I can get is getting the error (after moving a bunch of dlls in the same directory as the .exe

> > Failed to dispatch 'signal activate: \\.\E:\howl/lib/ljglibs/gtk/css_provider.moon:12: <data>:11:8not a number (code: 1)'
> Fontconfig error: Cannot load default config file
> 
> (howl.exe:5088): Gtk-WARNING **: Theme parsing error: <data>:11:39: Using Pango syntax for the font: style property is deprecated; please use CSS syntax

gcc also outputs 

> make: [Makefile:85: bytecode] Error 123 (ignored)
> make: *** [Makefile:86: bytecode] Error 123

during the end of the compile. 

Which I'm guessing is the reason the exe will not run.

:(


If you could upload a working binary I would appreciate!

**Comment by jstaursky** [comment]
okay I fixed the first wacky part "\.\E:\howl/lib/ljglibs/gtk/css_provider.moon:" because I realized I set up an environment variable wrong a while ago.
So instead I have

> Failed to dispatch 'signal activate: ../lib/ljglibs/gtk/css_provider.moon:12: <data>:11:8not a number (code: 1)'
> Fontconfig error: Cannot load default config file

same error I know but at least path seems sane.

This also fixes the gcc error and now the compiler outputs

>  Fontconfig error: Cannot load default config file

instead of Error 123.

However the final error appears to be the same when I try to run it :/

> Failed to dispatch 'signal activate: ../lib/ljglibs/gtk/css_provider.moon:12: <data>:11:8not a number (code: 1)'
> Fontconfig error: Cannot load default config file
> 
> (howl.exe:5348): Gtk-WARNING **: Theme parsing error: <data>:11:39: Using Pango syntax for the font: style property is deprecated; please use CSS syntax

**Comment by kirbyfan64** [comment]
@jstaursky https://mega.nz/#!ZBsGjJJB!ixyrSxq5Vec_G_2R_f9QBR20IbuUZWjInjJmiGGm7ss

**Comment by nilnor** [comment]
The CSS issue would likely be resolved if this branch was rebased on master (see 1fe80c52012dfe7dd1e7b646aba45e02da0dc781 for relevant fixing, but I guess just appending `px` to the font size [here](https://github.com/howl-editor/howl/blob/windows/lib/howl/ui/theme.moon#L79) might do it.

**Comment by kirbyfan64** [comment]
Yeah, I really need to rebase that, but I always manage to screw it up somehow... I was trying to see if I could fix the process launching issues first, but I'm still kind of stuck, though at least I've been able to narrow down the cause (coroutines).

I'll probably just end up making a new branch and cherry-picking all the old commits.

**Comment by jstaursky** [comment]
@kirbyfan64 Thanks for providing the binary, but I'm still running into trouble :(

> $ ./howl.exe
> E:\workinghowl\dist\share\howl\lib\howl\init.lua:188: ../lib/howl/application.moon:4: ../lib/howl/ui/window.moon:5: module 'ljglibs.aux.background' not found:
>         no field package.preload['ljglibs.aux.background']
>         no file '\\.\E:\workinghowl\dist\share\howl/lib/ljglibs\aux\background.lua'
>         no file '\\.\E:\workinghowl\dist\share\howl/lib/ljglibs\aux\background/init.lua'
>         no file '\\.\E:\workinghowl\dist\share\howl/lib/ext/ljglibs\aux\background.lua'
>         no file '\\.\E:\workinghowl\dist\share\howl/lib/ext/ljglibs\aux\background/init.lua'
>         no file '\\.\E:\workinghowl\dist\share\howl/lib/ext/moonscript/ljglibs\aux\background.lua'
>         no file '\\.\E:\workinghowl\dist\share\howl/lib/ext/moonscript/ljglibs\aux\background/init.lua'
>         no file '.\ljglibs\aux\background.lua'
>         no file 'E:\workinghowl\dist\bin\lua\ljglibs\aux\background.lua'
>         no file 'E:\workinghowl\dist\bin\lua\ljglibs\aux\background\init.lua'
>         no file '.\ljglibs\aux\background.dll'
>         no file 'E:\workinghowl\dist\bin\ljglibs\aux\background.dll'
>         no file 'E:\workin
> ** (howl.exe:3532): CRITICAL **: Failed to run script: E:\workinghowl\dist\share\howl\lib\howl\init.lua:210: E:\workinghowl\dist\share\howl\lib\howl\init.lua:188: ../lib/howl/application.moon:4: ../lib/howl/ui/window.moon:5: module 'ljglibs.aux.background' not found:
>         no field package.preload['ljglibs.aux.background']
>         no file '\\.\E:\workinghowl\dist\share\howl/lib/ljglibs\aux\background.lua'
>         no file '\\.\E:\workinghowl\dist\share\howl/lib/ljglibs\aux\background/init.lua'
>         no file '\\.\E:\workinghowl\dist\share\howl/lib/ext/ljglibs\aux\background.lua'
>         no file '\\.\E:\workinghowl\dist\share\howl/lib/ext/ljglibs\aux\background/init.lua'
>         no file '\\.\E:\workinghowl\dist\share\howl/lib/ext/moonscript/ljglibs\aux\background.lua'
>         no file '\\.\E:\workinghowl\dist\share\howl/lib/ext/moonscript/ljglibs\aux\background/init.lua'
>         no file '.\ljglibs\aux\background.lua'
>         no file 'E:\workinghowl\dist\bin\lua\ljglibs\aux\background.lua'
>         no file 'E:\workinghowl\dist\bin\lua\ljglibs\aux\background\init.lua'
>         no file '.\ljglibs\aux\background.dll'
>         no file 'E:\workinghowl\dist\bin\ljglibs\aux\background.dll'
>         no file 'E:\workinghowl\dist\bin\loadall.dll'
>         no file '.\ljglibs.dll'
>         no file 'E:\workinghowl\dist\bin\ljglibs.dll'
>         no file 'E:\workinghowl\dist\bin\loadall.dll'
> 
> ghowl\dist\bin\loadall.dll'
>         no file '.\ljglibs.dll'
>         no file 'E:\workinghowl\dist\bin\ljglibs.dll'
>         no file 'E:\workinghowl\dist\bin\loadall.dll'

At one point I had my own build getting something resembling these errors as well (not sure if it was asking for the same dlls) not sure what load "loadall.dll" is, I have searched my entire msys2 folder and couldn't find it.

@nilnor I'll give this a shot when I get back home from work. (running all this on my flash drive making me feel like a bad one).

**Comment by jstaursky** [comment]
No luck @nilnor I tried both fixes still ending up with 

> Failed to dispatch 'signal activate: ../lib/ljglibs/gtk/css_provider.moon:12: <data>:11:8not a number (code: 1)'
> Fontconfig error: Cannot load default config file
> 
> (howl.exe:12108): Gtk-WARNING **: Theme parsing error: <data>:11:39: Using Pango syntax for the font: style property is deprecated; please use CSS syntax
> 
> (howl.exe:12108): Gtk-WARNING **: Theme parsing error: <data>:16:39: Using Pango syntax for the font: style property is deprecated; please use CSS syntax
> 
> (howl.exe:12108): Gtk-WARNING **: Theme parsing error: <data>:20:46: Using Pango syntax for the font: style property is deprecated; please use CSS syntax
> 
> (howl.exe:12108): Gtk-WARNING **: Theme parsing error: <data>:23:65: Using Pango syntax for the font: style property is deprecated; please use CSS syntax
> 
> (howl.exe:12108): Gtk-WARNING **: Theme parsing error: <data>:24:49: Using Pango syntax for the font: style property is deprecated; please use CSS syntax
> 
> (howl.exe:12108): Gtk-WARNING **: Theme parsing error: <data>:25:52: Using Pango syntax for the font: style property is deprecated; please use CSS syntax
> 
> (howl.exe:12108): Gtk-WARNING **: Theme parsing error: <data>:27:46: Using Pango syntax for the font: style property is deprecated; please use CSS syntax
> 
> (howl.exe:12108): Gtk-WARNING **: Theme parsing error: <data>:31:46: Using Pango syntax for the font: style property is deprecated; please use CSS syntax
> 
> (howl.exe:12108): Gtk-WARNING **: Theme parsing error: <data>:35:46: Using Pango syntax for the font: style property is deprecated; please use CSS syntax
> 

When I use my .exe

with @kirbyfan64  binary it looks like I am just missing a lua module or something. 

Also I noticed there was no luajit in the attachment, I tried getting it to work with the luajit I could build by adding it to path but that didn't do anything.

**Comment by jstaursky** [comment]
I fixed it! Thank you @kirbyfan64 for the binary! 
All I had to do to fix my earlier problem was 
`cd /e/workinghowl/dist/share/howl/lib/ljglibs/`
`mkdir aux`
`cd _aux`
`cp background.bc ../aux/`
`cp background.moon ../aux/`

Here is my result after going back and running `./howl.exe`
![image](https://cloud.githubusercontent.com/assets/25406664/22610649/6fbd2326-ea34-11e6-816e-acc918bff269.png)
 Woo hoo!

**Comment by tuffnerdstuff** [comment]
Sorry for reviving this thread, but I ran into the same problem as @jstaursky while compiling howl for windows. When running howl I always got the following exception:

> Failed to dispatch 'signal activate: ../lib/ljglibs/gtk/css_provider.moon:12: :11:8not a number (code: 1)'

I dug around in the code and learned that css_provider.moon hands over CSS stylesheets to GTK and I read from the error message that something must be wrong with the CSS. After some more digging and debugging I figured out that theme.moon seems to format font styles in a wrong way. I looked up how to do it properly ([CSS font property](https://www.w3schools.com/cssref/pr_font_font.asp)), rearranged the way the font attributes are ordered and voila: howl started!

Of course I would like to share my (humble) patch with you, but I wanted to ask first if somebody knows why this seems to be only a problem in the windows build? I assume that GTK might have changed the way it parses CSS, but then it would be also a problem for the linux build.

Cheers
